packages/react-native-sdk/src/components/participants/ParticipantView.tsx

```diff
@@ -0,0 +1,276 @@
+      pendingVideoLayoutRef.current = subscribedVideoLayoutRef.current;
+      subscribedVideoLayoutRef.current = undefined;
+    }
+  }, [hasJoinedCall]);
+
+  /**
+   * This effect updates the subscription either
+   * 1. when video tracks are published and was unpublished before
+   * 2. when the view's visibility changes
+   * 3. when call was rejoined
+   */
+  useEffect(() => {
+    // NOTE: We only want to update the subscription if the pendingVideoLayoutRef is set
+    const updateIsNeeded = pendingVideoLayoutRef.current;
+    if (!updateIsNeeded || !call || !isPublishingVideoTrack || !hasJoinedCall) {
+      return;
+    }
+
+    // NOTE: When the view is not visible, we want to subscribe to audio only.
+    // We unsubscribe their video by setting the dimension to undefined
+    const dimension = !muteVideo ? pendingVideoLayoutRef.current : undefined;
+
+    call.updateSubscriptionsPartial(videoMode, {
+      [sessionId]: { dimension },
+    });
+
+    if (dimension) {
+      subscribedVideoLayoutRef.current = pendingVideoLayoutRef.current;
+      pendingVideoLayoutRef.current = undefined;
+    }
+  }, [
+    call,
+    isPublishingVideoTrack,
+    videoMode,
+    sessionId,
+    muteVideo,
+    hasJoinedCall,
+  ]);
+
+  useEffect(() => {
+    return () => {
+      subscribedVideoLayoutRef.current = undefined;
+      pendingVideoLayoutRef.current = undefined;
+    };
+  }, [videoMode, sessionId]);
+
+  const onLayout: React.ComponentProps<typeof View>['onLayout'] = (event) => {
+    const dimension = {
+      width: Math.trunc(event.nativeEvent.layout.width),
+      height: Math.trunc(event.nativeEvent.layout.height),
+    };
+
+    // NOTE: If the participant hasn't published a video track yet,
+    // or the view is not viewable, we store the dimensions and handle it
+    // when the track is published or the video is enabled.
+    if (!call || !isPublishingVideoTrack || muteVideo || !hasJoinedCall) {
+      pendingVideoLayoutRef.current = dimension;
+      return;
+    }
+
+    // NOTE: We don't want to update the subscription if the dimension hasn't changed
+    if (
+      subscribedVideoLayoutRef.current?.width === dimension.width &&
+      subscribedVideoLayoutRef.current?.height === dimension.height
+    ) {
+      return;
+    }
+
+    call.updateSubscriptionsPartial(videoMode, {
+      [sessionId]: {
+        dimension,
+      },
+    });
+    subscribedVideoLayoutRef.current = dimension;
+    pendingVideoLayoutRef.current = undefined;
+  };
+
+  const isScreenSharing = videoMode === 'screen';
+  const applySpeakerStyle = isSpeaking && !isScreenSharing;
+  const speakerStyle = applySpeakerStyle && styles.isSpeaking;
+
+  return (
+    <View
+      style={[styles.containerBase, props.containerStyle, speakerStyle]}
+      testID={`participant-${userId}`}
+      accessibilityValue={{
+        text: isSpeaking
+          ? 'participant-is-speaking'
+          : 'participant-is-not-speaking',
+      }}
```

Comment:
can we remove this? is it used for testing?

Reply:
Yes its used for testing.

Reply:
@khushal87 lets remove it! move this to testid

Reply:
We also have a test id here 😅

Reply:
```
testID={isSpeaking ? `participant-${userId}-is-speaking` : `'participant-${userId}-is-not-speaking'`}
```

Reply:
or 
```
testID={`participant-${userId}-is-${isSpeaking ? "" : "not-"}speaking`
```

Reply:
Yeah, that's the only solution